test/Test/Mafia/IO.hs

```diff
@@ -12,10 +11,7 @@ import           Test.QuickCheck.Monadic
 import           System.CPUTime (getCPUTime)
 
 testIO :: Testable a => IO a -> Property
-testIO = testPropertyIO . run
-
-testPropertyIO :: Testable a => PropertyM IO a -> Property
-testPropertyIO = monadicIO . (=<<) stop
+testIO = monadicIO . run
```

Comment:
Not sure this is doing the right thing, bumped the QuickCheck version min to 2.10 and it seems to work locally.

Reply:
That seems reasonably sane.

Did you force push this branch because it doesn't seem to have my commits?

Reply:
Can probably avoid bumping the min version by writing

```haskell
testIO k = monadicIO $ do
  p <- run k
  stop p
  return p
```

Reply:
@erikd this is based off my fork and branch. Where did you make your changes?

@thumphries thanks :-)

Reply:
I pushed my changes to the `topic/ghc_next` branch here on git hub.